src/app/[locale]/page.tsx

```diff
@@ -156,14 +194,37 @@ const Profile = () => {
                   </Tooltip>
                 </Grid>
               </Grid>
+              <Divider />
+              {walletData?.status !== 'installed' && (
```

Comment:
what about moving all this fragment and the above `useEffect()` logic into its own component?

we should have here (in page.tsx) only something like:

`<HomeWalletCard isProfileAvailable={isProfileAvailable} isWalletRevocationActive={isWalletRevocationActive} />
`